File: src/main/java/io/saso/dash/services/SubServiceAdapter.java
package io.saso.dash.services;

import io.netty.channel.ChannelHandlerContext;

public abstract class SubServiceAdapter implements ServicePollable
{
    @Override
    public void start(ChannelHandlerContext ctx, DBEntityProvider db)
            throws Exception
    { /* empty */ }

    @Override
    public void poll(ChannelHandlerContext ctx, DBEntityProvider db)
            throws Exception
    { /* empty */ }

    @Override
    public void stop(ChannelHandlerContext ctx, DBEntityProvider db)
            throws Exception
    { /* empty */ }
}

File: src/main/kotlin/io/saso/dash/services/SubServiceAdapter.kt
package io.saso.dash.services

import io.netty.channel.ChannelHandlerContext
import io.netty.handler.codec.http.websocketx.TextWebSocketFrame
import io.saso.dash.database.EntityProvider

public abstract class SubServiceAdapter : ServicePollable
{
    override fun start(ctx: ChannelHandlerContext, db: EntityProvider) { }

    override fun poll(ctx: ChannelHandlerContext, db: EntityProvider) { }

    override fun stop(ctx: ChannelHandlerContext, db: EntityProvider) { }

    protected fun write(ctx: ChannelHandlerContext, toWrite: String)
    {
        ctx.channel().writeAndFlush(TextWebSocketFrame(toWrite))
    }
}
